src/frontend/packages/lib_components/src/hooks/stores/useJwt/index.ts

```diff
@@ -10,11 +10,12 @@ interface JwtStoreInterface {
   jwt?: string;
   refreshJwt?: string;
   internalDecodedJwt?: DecodedJwt;
-  getJwt: () => string | undefined;
-  setJwt: (jwt: string) => void;
-  getRefreshJwt: () => string | undefined;
-  setRefreshJwt: (jwt: string) => void;
-  getDecodedJwt: () => DecodedJwt;
+  getJwt: () => JwtStoreInterface['jwt'];
+  setJwt: (jwt?: JwtStoreInterface['jwt']) => void;
+  getRefreshJwt: () => JwtStoreInterface['refreshJwt'];
+  setRefreshJwt: (refreshJwt: JwtStoreInterface['refreshJwt']) => void;
+  setDecodedJwt: (jwt: JwtStoreInterface['jwt']) => void;
```

Comment:
IMO I understand I have to set the decoded jwt. Not to use the `jwt` at parameter and the function should decode it for us.

Reply:
Should we decode it in the `setJwt` method ? Doing this, the jwt and the decoded one are always in sync.

Reply:
Actually it is like that, but I had it in this commit: [(website) update permission during jwt init](https://github.com/openfun/marsha/pull/2244/commits/be245ef727e6a98bdf0550fa484a0cb8c6ef50fe#diff-f6fc6375622192706059fba50547c99d0518c270c862829dbdc1f44b2fdcaec9R65)